lib/widget/roles_selection.dart

```diff
@@ -0,0 +1,211 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:heist/app_localizations.dart';
+import 'package:heist/colors.dart';
+import 'package:heist/middleware/room_middleware.dart';
+import 'package:heist/role.dart';
+import 'package:heist/selectors/selectors.dart';
+import 'package:heist/state.dart';
+import 'package:heist/widget/common.dart';
+import 'package:redux/redux.dart';
+
+class RolesSelection extends StatefulWidget {
+  final Store<GameModel> _store;
+
+  RolesSelection(this._store);
+
+  @override
+  State<StatefulWidget> createState() => RolesSelectionState(_store);
+}
+
+class RolesSelectionState extends State<RolesSelection> {
```

Comment:
Tip: I realised the other day that if you name this class `_RolesSelectionState`, then it will be a 'private' class and you don't have to bother with underscoring every variable and method in it.

Reply:
I've underscored the class and ununderscored the methods.